### PR TITLE
Make question title optional and display the question number

### LINKED ISSUE #1218: Make question title optional and display as Question N when there's no title

### PR DESCRIPTION
**Pull request by fonglh**
Fixes #1218

### CODE REVIEW

File: app/models/course/assessment/question.rb [hunk 1]
```diff
@@ -56,6 +56,17 @@ def last_question?
...
 
+  # Prefixes a question number in front of the title
+  #
+  # @return [string]
+  def display_title
+    question_number = I18n.t('activerecord.course/assessment/question.question_number',
+                             index: assessment.questions.index(self) + 1)
+
+    return question_number if title.blank?
+    "#{question_number}: #{title}"
```
Comment (allenwq) [inline app/models/course/assessment/question.rb]: Actually the `:` might be different in some other languages

File: spec/models/course/assessment/question_spec.rb [hunk 2]
```diff
@@ -41,6 +41,28 @@ def self.table_name
...
     end
 
+    describe '#display_title' do
+      context 'when title is nil' do
+        let!(:question) { create(:course_assessment_question, title: nil) }
+        subject { question }
+
+        it 'returns Question N' do
+          expect(subject.display_title).
+            to eq('activerecord.course/assessment/question.question_number')
```
Comment (allenwq) [inline spec/models/course/assessment/question_spec.rb]: `I18n.t('activerecord.course/assessment/question.question_number')`

File: app/models/course/assessment/question.rb (hunk 1 above)
Comment (allenwq) [inline app/models/course/assessment/question.rb]: Maybe you should use 2 translations instead, one with `:`, what do you think ?

Comment (fonglh) [inline app/models/course/assessment/question.rb]: Ok I'll try that out.

**Pull request merged**